# AMAZON TOP 50 BESTSELLING BOOKS



<br/>

In [1]:
#Importing packages

import pandas as pd
import numpy as np

#Data Visualization 

import matplotlib.pyplot as plt
plt.style.use('ggplot')
import seaborn as sns
sns.set_palette('husl')

import plotly.express as px
import plotly.graph_objs as go
import plotly.offline as pyo
from plotly.subplots import make_subplots

# Special Visualization
import wordcloud, missingno
from wordcloud import WordCloud # wordcloud
import missingno as msno # check missing value
import networkx as nx

'''Plotly visualization .'''
import plotly.offline as py
from plotly.offline import iplot, init_notebook_mode
import plotly.graph_objs as go
py.init_notebook_mode(connected = True) # Required to use plotly offline in jupyter notebook

import cufflinks as cf #importing plotly and cufflinks in offline mode  
import plotly.offline  
cf.go_offline()  
cf.set_config_file(offline=False, world_readable=True)

from bubbly.bubbly import bubbleplot 
from termcolor2 import c

### Reading and quick overview of the data

In [2]:
#Import data

data = pd.read_csv("bestsellers with categories.csv")
data.head(10)


Name  \
0                      10-Day Green Smoothie Cleanse   
1                                  11/22/63: A Novel   
2            12 Rules for Life: An Antidote to Chaos   
3                             1984 (Signet Classics)   
4  5,000 Awesome Facts (About Everything!) (Natio...   
5      A Dance with Dragons (A Song of Ice and Fire)   
6  A Game of Thrones / A Clash of Kings / A Storm...   
7                     A Gentleman in Moscow: A Novel   
8      A Higher Loyalty: Truth, Lies, and Leadership   
9                          A Man Called Ove: A Novel   

                     Author  User Rating  Reviews  Price  Year        Genre  
0                  JJ Smith          4.7    17350      8  2016  Non Fiction  
1              Stephen King          4.6     2052     22  2011      Fiction  
2        Jordan B. Peterson          4.7    18979     15  2018  Non Fiction  
3             George Orwell          4.7    21424      6  2017      Fiction  
4  National Geographic Kids          4.8     7665     12  2019  Non Fiction  
5       George R. R. Martin          4.4    12643     11  2011      Fiction  
6       George R. R. Martin          4.7    19735     30  2014      Fiction  
7               Amor Towles          4.7    19699     15  2017      Fiction  
8               James Comey          4.7     5983      3  2018  Non Fiction  
9           Fredrik Backman          4.6    23848      8  2016      Fiction

In [3]:
data.tail(10)

Name         Author  \
540                                             Wonder  R. J. Palacio   
541                                             Wonder  R. J. Palacio   
542                                             Wonder  R. J. Palacio   
543                                             Wonder  R. J. Palacio   
544                                             Wonder  R. J. Palacio   
545       Wrecking Ball (Diary of a Wimpy Kid Book 14)    Jeff Kinney   
546  You Are a Badass: How to Stop Doubting Your Gr...    Jen Sincero   
547  You Are a Badass: How to Stop Doubting Your Gr...    Jen Sincero   
548  You Are a Badass: How to Stop Doubting Your Gr...    Jen Sincero   
549  You Are a Badass: How to Stop Doubting Your Gr...    Jen Sincero   

     User Rating  Reviews  Price  Year        Genre  
540          4.8    21625      9  2013      Fiction  
541          4.8    21625      9  2014      Fiction  
542          4.8    21625      9  2015      Fiction  
543          4.8    21625      9  2016      Fiction  
544          4.8    21625      9  2017      Fiction  
545          4.9     9413      8  2019      Fiction  
546          4.7    14331      8  2016  Non Fiction  
547          4.7    14331      8  2017  Non Fiction  
548          4.7    14331      8  2018  Non Fiction  
549          4.7    14331      8  2019  Non Fiction

In [4]:
data.describe(include='all')

Name       Author  \
count                                                 550          550   
unique                                                351          248   
top     Publication Manual of the American Psychologic...  Jeff Kinney   
freq                                                   10           12   
mean                                                  NaN          NaN   
std                                                   NaN          NaN   
min                                                   NaN          NaN   
25%                                                   NaN          NaN   
50%                                                   NaN          NaN   
75%                                                   NaN          NaN   
max                                                   NaN          NaN   

        User Rating       Reviews       Price         Year        Genre  
count    550.000000    550.000000  550.000000   550.000000          550  
unique          NaN           NaN         NaN          NaN            2  
top             NaN           NaN         NaN          NaN  Non Fiction  
freq            NaN           NaN         NaN          NaN          310  
mean       4.618364  11953.281818   13.100000  2014.000000          NaN  
std        0.226980  11731.132017   10.842262     3.165156          NaN  
min        3.300000     37.000000    0.000000  2009.000000          NaN  
25%        4.500000   4058.000000    7.000000  2011.000000          NaN  
50%        4.700000   8580.000000   11.000000  2014.000000          NaN  
75%        4.800000  17253.250000   16.000000  2017.000000          NaN  
max        4.900000  87841.000000  105.000000  2019.000000          NaN

In [5]:
data.describe()

User Rating       Reviews       Price         Year
count   550.000000    550.000000  550.000000   550.000000
mean      4.618364  11953.281818   13.100000  2014.000000
std       0.226980  11731.132017   10.842262     3.165156
min       3.300000     37.000000    0.000000  2009.000000
25%       4.500000   4058.000000    7.000000  2011.000000
50%       4.700000   8580.000000   11.000000  2014.000000
75%       4.800000  17253.250000   16.000000  2017.000000
max       4.900000  87841.000000  105.000000  2019.000000

In [6]:
data.describe(include='O')

Name       Author  \
count                                                 550          550   
unique                                                351          248   
top     Publication Manual of the American Psychologic...  Jeff Kinney   
freq                                                   10           12   

              Genre  
count           550  
unique            2  
top     Non Fiction  
freq            310

# BestSellers Ratings

In [7]:
user_ratings = data["User Rating"].value_counts().reset_index()
user_ratings

trace = go.Bar(
                x = user_ratings["index"],
                y = user_ratings["User Rating"],
                marker = dict(color = "#0C619D",
                        line= dict(color = "#FFFAF0" , width =1.7)))
chart_layout = go.Layout(template = "plotly_dark", title = "BESTSELLERS RATINGS", xaxis =dict(title = "Ratings") , yaxis = dict(title = "Counts"))

fig = go.Figure(data = [trace] , layout = chart_layout)
fig.show()


def pie_plot(cnt_srs,colors, title):
    labels=cnt_srs.index
    values=cnt_srs.values
    trace = go.Pie(labels=labels, 
                   values=values, 
                   title=title, 
                   hoverinfo='percent+value+label', 
                   textinfo='percent',
                   textposition='inside',
                   hole=0.5,
                   showlegend=True,
                   marker=dict(colors=colors,
                               line=dict(color='#FFFAF0',
                                         width=2.7),
                              )
                  )
    return trace
print ("**MOST OF THE RATINGS ARE IN THE RANGE OF 4.6 TO 4.8**")


py.iplot([pie_plot(data['User Rating'].value_counts(),['orange red',"brown","deep sky blue"], 'Ratings')])


**MOST OF THE RATINGS ARE IN THE RANGE OF 4.6 TO 4.8**


<span style="font-family:Lucida Sans">We can see the percentage share of each ratings, with 4.8 taking 23.1% of the total share</span>

 <br />

# Relationship between the Price and ratings of each user

In [8]:
#
trace = go.Bar(
                x = data["Price"],
                y = data["User Rating"],
                marker = dict(color = "#0C619D",
                        line= dict(color = "#FFFAF0" , width =0.1)))
chart_layout = go.Layout(template = "plotly_dark", title = "Price and ratings", xaxis =dict(title = "Price") , yaxis = dict(title = "User Rating"))

fig = go.Figure(data = [trace] , layout = chart_layout)
fig.show()


<span style="font-family:Lucida Sans;font-size:30"> From the above chart, we could notice that most of the books rated are well below 20 dollars, with the highest at 8 dollars, would it then be safe to say that most of the books purchased are less that 20 dollars?</span>


 <br />

 <br />

<span style="font-family:Comic Sans MS"></span>

<span style="font-family:Lucida Sans"></span>

# Relationship between the Price and Reviews

In [9]:
#
trace = go.Bar(
                x = data["Price"],
                y = data["Reviews"],
                marker = dict(color = "#0C619D",
                        line= dict(color = "#FFFAF0" , width =0.1)))
chart_layout = go.Layout(template = "plotly_dark", title = "Price and Reviews", xaxis =dict(title = "Price") , yaxis = dict(title = "Reviews"))

fig = go.Figure(data = [trace] , layout = chart_layout)
fig.show()

<span style="font-family:Lucida Sans">There is a similar trend with the "Price and Reviews" chart as seen also in the "Price and Ratings" chart. This doesnt mean that the reviews are positive though or even negative . </span> </p> <span style="font-family:Lucida Sans">It just shows that people are willing to drop reviews for the books they bought. </span>

 <br />

# GENRES

In [10]:
unique_genre = data["Genre"].nunique()
unique_genre

2

<span style="font-family:Lucida Sans"> The books are classified into 2 genres, which are Fiction and Non Fiction </span><br/>

<strong><span style="font-family:Lucida Sans">Which genre would you guess to be the most popular? Fiction or Non Fiction? Let's find out!!!</span><strong>

In [11]:
genre = data["Genre"].value_counts().reset_index()
genre

fiction = data[data["Genre"] == "Fiction"]
fiction = fiction["Genre"].value_counts().reset_index()


non_fiction = data[data["Genre"] == "Non Fiction"]
non_fiction = non_fiction["Genre"].value_counts().reset_index()


trace_fiction = go.Bar(
                x = fiction['index'],
                y = fiction['Genre'],
                name = "Fiction",
                marker = dict(color = "#1FC3AA",
                        line= dict(color = "#FFFAF0" , width =2.5)))

trace_non_fiction = go.Bar(
                x = non_fiction['index'],
                y = non_fiction['Genre'],
                name = "Non fiction",
                marker = dict(color = "#8624F5",
                        line = dict(color = "#FFFAF0", width = 2.5)))

layouts = go.Layout(template = "plotly_dark" ,title = "Fiction and Non fiction", xaxis = dict(title = "Counts") , yaxis = dict(title = "Genre"))

fig = go.Figure(data = [trace_fiction ,trace_non_fiction], layout = layouts)
fig.show()


py.iplot([pie_plot(data['Genre'].value_counts(),['#8624F5', '#1FC3AA'], 'Genre')])

<span style="font-family:Lucida Sans"> Majority of the bestselling books are Non Fiction, taking a total of 56.4%.</span>

<strong><span style="font-family:Lucida Sans"> I always thought Fiction would be the most popular genre, but I guess people are more intrested in reality that tales.</span><strong>
<br><strong><span style="font-family:Lucida Sans">Since this data is based on a limited timeframe and not an all time data, lets find out which genre was popular in each year</span><strong> 

# Genre Popularity Per Year 

In [12]:

fic= data[data["Genre"] == "Fiction"]
fict = fic["Year"].value_counts().reset_index()
fict = fict.rename(columns = {"index" : "Year" , "Year" : "Count"})
fict = fict.sort_values("Year")



non_fic= data[data["Genre"] == "Non Fiction"]
non_fict = non_fic["Year"].value_counts().reset_index()
non_fict = non_fict.rename(columns = {"index" : "Year" , "Year" : "Count"})
non_fict = non_fict.sort_values("Year")


trace_fict = go.Scatter(
                    x = fict['Year'],
                    y = fict["Count"],
                    name = "Fiction",
                    marker = dict(color = "#1FC3AA",
                        line= dict(color = "#FFFAF0" , width =2.5)))
    
trace_non_fict = go.Scatter(
                    x = non_fict['Year'],
                    y = non_fict["Count"],
                    name = "Non Fiction",
                    marker = dict(color = "#8624F5",
                        line= dict(color = "#FFFAF0" , width =2.5)))

layouts = go.Layout(template = "plotly_dark" ,title = "Genre Popularity Per Year", xaxis = dict(title = "Year") , yaxis = dict(title = "Count"))

fig = go.Figure(data = [trace_fict ,trace_non_fict], layout = layouts)
fig.show()

<span style="font-family:Lucida Sans">From the above graph,we can see that the total Bestselling books' count per year is 50. This also shows that Non fiction books were very much prefered especially in the year 2015</span>

<strong><span style="font-family:Lucida Sans">I'm wondering what could have caused the sudden rise and fall of Fictional bestsellers in 2014 and 2015 respectively. What external factor influenced such drastic change?But our current data can't give us any insight on that.<br>Lets find out how each genre was rated. Which genre had higher ratings? Did people actually enjoy the Non fictional books they purchased?</span><strong><p>


# Genre Ratings Per Year 

In [13]:

fictions = data[data["Genre"] == "Fiction"]
fictions = fictions["User Rating"].value_counts().reset_index()


non_fictions = data[data["Genre"] == "Non Fiction"]
non_fictions = non_fictions["User Rating"].value_counts().reset_index()


trace_fictions = go.Bar(
                x = fictions['index'],
                y = fictions['User Rating'],
                name = "Fiction",
                marker = dict(color = "#1FC3AA",
                        line= dict(color = "#FFFAF0" , width =1.5)))

trace_non_fictions = go.Bar(
                x = non_fictions['index'],
                y = non_fictions['User Rating'],
                name = "Non fiction",
                marker = dict(color = "#8624F5",
                        line = dict(color = "#FFFAF0", width = 1.5)))

layouts = go.Layout(template = "plotly_dark" ,title = "Genre Ratings Per Year ", xaxis = dict(title = "User Ratings") , yaxis = dict(title = "Counts"))

fig = go.Figure(data = [trace_fictions ,trace_non_fictions], layout = layouts)
fig.show()




<span style="font-family:Lucida Sans">Fictional Books were rated higher than non fictional books within the 4.8 and 4.9 only, otherwise Non Fiction was rated better. Also it is worthy to note that only fictional books were rated below 4</span>


<strong><span style="font-family:Lucida Sans">It seems people do actually like the Non fictional books as it was given a no bad rating unlike the fictional books</span><strong><p>